src/scripts/main.js

```diff
@@ -1,3 +1,172 @@
 'use strict';
 
 // write code here
+document.addEventListener('DOMContentLoaded', function () {
+  const employeeTable = document.querySelector('table');
+  let currentSort = { column: null, order: 'asc' };
+  let selectedRow = null;
+
+  const headers = document.querySelectorAll('th');
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      const order =
+        currentSort.column === index && currentSort.order === 'asc'
+          ? 'desc'
+          : 'asc';
+
+      sortTable(employeeTable, index, order);
+      currentSort = { column: index, order: order };
+    });
+  });
+
+  function sortTable(table, columnIndex, order) {
+    const rows = Array.from(table.tBodies[0].rows);
+    const isNumeric = columnIndex === 3 || columnIndex === 4;
+
+    rows.sort((rowA, rowB) => {
+      let a = rowA.cells[columnIndex].innerText.trim();
+      let b = rowB.cells[columnIndex].innerText.trim();
+
+      if (isNumeric) {
+        a = parseFloat(a.replace(/[^0-9.-]+/g, ''));
+        b = parseFloat(b.replace(/[^0-9.-]+/g, ''));
+      }
+
+      if (order === 'asc') {
+        return isNumeric ? a - b : a.localeCompare(b);
+      } else {
+        return isNumeric ? b - a : b.localeCompare(a);
+      }
+    });
+
+    rows.forEach((row) => table.tBodies[0].appendChild(row));
+  }
```

Comment:
This function is quite large and does a lot of things. Consider breaking it down into smaller, more manageable functions. For example, you could have separate functions for getting the cell values, comparing the values, and reordering the rows. This will make your code easier to read and test.